src/Bridges/Constraint/bridges/integer_to_zeroone.jl

```diff
@@ -0,0 +1,168 @@
+    ::Type{MOI.Integer},
+) where {T}
+    return true
+end
+
+function MOI.Bridges.added_constrained_variable_types(
+    ::Type{<:IntegerToZeroOneBridge},
+)
+    return Tuple{Type}[(MOI.ZeroOne,)]
+end
+
+function MOI.Bridges.added_constraint_types(
+    ::Type{IntegerToZeroOneBridge{T}},
+) where {T}
+    return Tuple{Type,Type}[(MOI.ScalarAffineFunction{T}, MOI.EqualTo{T})]
+end
+
+function concrete_bridge_type(
+    ::Type{IntegerToZeroOneBridge{T}},
+    ::Type{MOI.VariableIndex},
+    ::Type{MOI.Integer},
+) where {T}
+    return IntegerToZeroOneBridge{T}
+end
+
+function MOI.get(
+    ::MOI.ModelLike,
+    ::MOI.ConstraintFunction,
+    bridge::IntegerToZeroOneBridge,
+)
+    return bridge.x
+end
+
+function MOI.get(::MOI.ModelLike, ::MOI.ConstraintSet, ::IntegerToZeroOneBridge)
+    return MOI.Integer()
+end
+
+function MOI.delete(model::MOI.ModelLike, bridge::IntegerToZeroOneBridge)
+    MOI.delete(model, bridge.ci)
+    MOI.delete(model, bridge.y)
+    return
+end
+
+function MOI.get(bridge::IntegerToZeroOneBridge, ::MOI.NumberOfVariables)::Int64
+    return length(bridge.y)
+end
+
+function MOI.get(bridge::IntegerToZeroOneBridge, ::MOI.ListOfVariableIndices)
+    return copy(bridge.y)
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge,
+    ::MOI.NumberOfConstraints{MOI.VariableIndex,MOI.ZeroOne},
+)::Int64
+    return length(bridge.y)
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge,
+    ::MOI.ListOfConstraintIndices{MOI.VariableIndex,MOI.ZeroOne},
+)
+    return map(bridge.y) do y
+        return MOI.ConstraintIndex{MOI.VariableIndex,MOI.ZeroOne}(y.value)
+    end
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge{T},
+    ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{T},MOI.EqualTo{T}},
+)::Int64 where {T}
+    return 1
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge{T},
+    ::MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{T},MOI.EqualTo{T}},
+) where {T}
+    return [bridge.ci]
+end
+
+MOI.Bridges.needs_final_touch(::IntegerToZeroOneBridge) = true
+
+function MOI.Bridges.final_touch(
+    bridge::IntegerToZeroOneBridge{T},
+    model::MOI.ModelLike,
+) where {T}
+    ret = MOI.Utilities.get_bounds(model, T, bridge.x)
+    if ret === bridge.last_bounds
+        return nothing  # final_touch already called
+    elseif ret[1] == typemin(T) || ret[2] == typemax(T)
+        error(
+            "Unable to use IntegerToZeroOneBridge because the variable " *
+            "$(bridge.x) has a non-finite domain",
+        )
+    end
+    f = MOI.ScalarAffineFunction([MOI.ScalarAffineTerm(T(1), bridge.x)], T(0))
+    lb, ub = ceil(Int, ret[1]), floor(Int, ret[2])
+    N = floor(Int, log2(ub - lb)) + 1
+    for i in 1:N
+        y, _ = MOI.add_constrained_variable(model, MOI.ZeroOne())
+        push!(bridge.y, y)
+        push!(f.terms, MOI.ScalarAffineTerm(-(T(2)^(i - 1)), y))
+    end
+    bridge.ci = MOI.add_constraint(model, f, MOI.EqualTo{T}(lb))
```

Comment:
The variable bridge spirit would be to substitute `x`. Here it seems to be a bad idea because it `x` is a  big sum.
This is like bridge a constrained variables `x in Int` by first adding the variable free and constraining it, then functionizing it and then slack bridge and then variable bridge for the slack. I'm wondering if there could be a nice way to implement this as a variable bridge but defaulting it to be used like that while allowing the user to have `x` be substituted as well.

Reply:
Could you just drop a comment that it would more naturally be implemented as a variable bridge but we don't want to substitute every occurence of `x` so that's why we do it as a constraint bridge ?

Reply:
> I'm wondering if there could be a nice way to implement this as a variable bridge

That would only be necessary for a solver supports _only_ binary variables and not real. We don't really have any current examples, although @ccoffrin might be interested for future quantum annealing formulations. (They have some pretty specific structure restrictions though, which mean doing it naively might be the wrong thing to do.)